src/order/bounded_order.lean

```diff
@@ -939,17 +1024,27 @@ lemma not_top_le_coe (a : α) : ¬ (⊤ : with_top α) ≤ ↑a := with_bot.not_
 lemma le_coe : ∀ {o : option α}, a ∈ o → (@has_le.le (with_top α) _ o b ↔ a ≤ b) | _ rfl :=
 coe_le_coe
 
-lemma le_coe_iff : ∀ {x : with_top α}, x ≤ b ↔ ∃ a : α, x = a ∧ a ≤ b
-| (some a) := by simp [some_eq_coe, coe_eq_coe]
-| none     := iff_of_false (not_top_le_coe _) $ by simp [none_eq_top]
+lemma le_coe_iff {x : with_top α} : x ≤ b ↔ ∃ a : α, x = a ∧ a ≤ b :=
+begin
+  simp only [←to_dual_le_to_dual_iff, to_dual_apply_coe, with_bot.coe_le_iff, order_dual.exists,
+             to_dual_le_to_dual],
+  exact exists_congr (λ _, and_congr_left' iff.rfl)
+end
```

Comment:
Copy-paste accident?

Reply:
What do you mean?

Reply:
Why did this proof get longer?

Reply:
shortened it even more than the original, without breaking through to the option API.